### PR TITLE
Update memberlist leave timeout and `max_global_series_per_metric` in all plans.

### PR DESCRIPTION
**Pull request by pstibrany**
#### What this PR does

This PR updates `operations/mimir` jsonnet to match production values used in Grafana Labs environment:
* Change `max_global_series_per_metric` to 0 in all plans, and as a default value.
* Memberlist: Increase the leave timeout to 10x the connection timeout, so that we can communicate the leave to at least 1 node, if the first 9 we try to contact times out.

#### Checklist

- [x] Tests updated
- [na] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -75,6 +75,8 @@
     - `query_frontend_topology_spread_max_skew`
     - `querier_topology_spread_max_skew`
     - `ruler_topology_spread_max_skew`
+* [CHANGE] Change `max_global_series_per_metric` to 0 in all plans, and as a default value.
+* [CHANGE] Memberlist: Increase the leave timeout to 10x the connection timeout, so that we can communicate the leave to at least 1 node, if the first 9 we try to contact times out.
```
Comment (pracucci) [inline CHANGELOG.md:79]: Can you add the PR number, please?

File: operations/mimir/memberlist.libsonnet [hunk 2]
```diff
@@ -45,6 +45,11 @@
     memberlistConfig:: {
       'memberlist.bind-port': gossipRingPort,
       'memberlist.join': 'dns+gossip-ring.%s.svc.cluster.local:%d' % [$._config.namespace, gossipRingPort],
+
+      // Increase the leave timeout to being 10x the connection timeout, so that we can communicate
+      // the leave to at least 1 node, if the first 9 we try to contact times out.
+      'memberlist.leave-timeout': '20s',
+      'memberlist.packet-dial-timeout': '2s',
```
Comment (pracucci) [inline operations/mimir/memberlist.libsonnet:52]: I'm fine to do this in Jsonnet first, but I think we should really change the default config in the code.

**Review by pracucci — APPROVED** [review]
LGTM, but see my comments. Thanks!

File: operations/mimir/memberlist.libsonnet (hunk 2 above)
Comment (pstibrany) [reply operations/mimir/memberlist.libsonnet:52]: Let's do that. https://github.com/grafana/dskit/pull/209

Comment (pstibrany) [reply operations/mimir/memberlist.libsonnet]: Done, ptal.

**Review by pracucci — APPROVED** [review]
LGTM, thanks!

**Pull request merged**